consensus/scheme/rolldpos/rolldpos.go

```diff
@@ -236,6 +236,15 @@ func (r *RollDPoS) CurrentState() fsm.State {
 	return r.cfsm.CurrentState()
 }
 
+// Activate activates or pauses the roll-DPoS consensus. When it is deactivated, the node will finish the current
+// consensus round if it is doing the work and then return the the initial state
+func (r *RollDPoS) Activate(active bool) { r.ctx.Activate(active) }
+
+// Active is true if the roll-DPoS consensus is active, or false if it is stand-by
+func (r *RollDPoS) Active() bool {
+	return r.ctx.Active() || r.cfsm.CurrentState() != consensusfsm.InitState
```

Comment:
what's the second part for?

Reply:
If the consensus fsm is not at init state, it's also considered as active